### PR TITLE
Add separate package for modular UI components

### PR DESCRIPTION
**Pull request by marthacryan**
In anticipation of creating a similar extension (see #186) to code snippets, 
this moves code that's already very modular into a separate package
called ui-components. To start, only ExpandableComponents.tsx is
moved, but moving forward, more will be added to the ui-components
package.
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: packages/ui-components/style/index.css [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+.elyra-expandableContainer-actionButton:hover {
+  background-color: #ffffff;
+}
+
+.elyra-expandableContainer-actionButton:active {
+  background-color: #eeeeee;
+}
+
+.elyra-codeSnippetsHeader {
```
Comment (ajbozarth) [inline packages/ui-components/style/index.css:36]: we shouldn't have any css with  code snippet in the name in this package, it should either be moved by to code snippet's css file or renamed

**Comment by marthacryan** [comment]
> one small comment below. On a different note, based on a comment from @lresende in one of my prs we should actually also move `icons.tsx` out of application and into ui-components as well.

@ajbozarth Could you elaborate?

**Comment by marthacryan** [comment]
@ajbozarth Does this look ok?

**Comment by ajbozarth** [comment]
At a glance that looks good, I would want to double check that you didn't miss any references to be sure though

**Comment by vabarbosa** [comment]
Related issue: https://github.com/elyra-ai/elyra/issues/403

**Comment by ajbozarth** [comment]
@marthacryan I fixed the merge conflict (via the ui) in the Makefile caused by my build update being merged. We should double check this still builds successfully, otherwise this is good to merge

**Comment by marthacryan** [comment]
@ajbozarth Just tried locally and it worked for me!

**Comment by ajbozarth** [comment]
@marthacryan great! Given the importance of this pr I'll defer to @lresende for final merge since he hasn't given a review yet

File: Makefile [hunk 2]
```diff
@@ -45,6 +45,7 @@ purge:
 
 uninstall:
 	$(call UNLINK_LAB_EXTENSION,@elyra/application)
+	$(call UNLINK_LAB_EXTENSION,@elyra/ui-components)
```
Comment (lresende) [inline Makefile:48]: @ajbozarth should this be **link/unlink** or **install/uninstall** ?

Comment (ajbozarth) [reply Makefile:48]: link/unlink, like application ui-components is a library not an extension

Comment (lresende) [reply Makefile:48]: makes sense, I thought it was actually an extension.

**Pull request merged**